t/12_sles4sap_publiccloud.t

```diff
@@ -535,6 +537,50 @@ subtest '[get_hana_topology]' => sub {
     ok((any { qr/SAPHanaSR-showAttr --format=/ } @calls), 'function calls SAPHanaSR-showAttr');
 };
 
+subtest '[get_hana_topology_json]' => sub {
+    my @calls;
+    my $self = sles4sap_publiccloud->new();
+    set_var('USE_SAP_HANA_SR_ANGI', 'true');
+    my $sles4sap_publiccloud = Test::MockModule->new('sles4sap_publiccloud', no_auto => 1);
+    $sles4sap_publiccloud->redefine(record_info => sub { note(join(' ', 'RECORD_INFO -->', @_)); });
+    my %test_topology = (
+        Host => {
+            vmhanaBBBBB => {
+                vhost => 'vmhanaBBBBB',
+                site => 'site_b'
+            },
+            vmhanaAAAAA => {
+                site => 'site_a',
+                vhost => 'vmhanaAAAAA',
+            }
+        },
+        Site => {
+            site_b => {
+                lss => '4',
+                mns => 'vmhanaAAAAA',
+                srPoll => 'SOK',
+            },
+            site_a => {
+                lss => '4',
+                mns => 'vmhanaBBBBB',
+                srPoll => 'PRIM',
+            }
+        }
+    );
+    $sles4sap_publiccloud->redefine(calculate_hana_topology => sub { return encode_json \%test_topology; });
```

Comment:
I do not understand this one. From these 2 lines of merged PR  

https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/21501/files#diff-4a40a74cf6048efd08f8d55e2613d752fb683e6dc56fc331da7e8f23ac0af1e9R181-R182

I understand the real `calculate_hana_topology` return a perl HASH not a json string. So I'd expect the same to be here:

```suggestion
    $sles4sap_publiccloud->redefine(calculate_hana_topology => sub { return \%test_topology; });
```

Reply:
Did you spot that is separate test for JASON inpuit to hana topology, right?